File: my-folder/0014-longest-common-prefix/solution.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
    
    if(strs.length==0){
        return "";
    }
      String prev=strs[0];
      for(int i=1;i<strs.length;i++){
          String curr=strs[i];
          while(curr.indexOf(prev)!=0){
              prev=prev.substring(0,prev.length()-1);
          }
      }
    return prev;
    }
}

File: my-folder/0014-longest-common-prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        # Normal iteration
        prefix = strs[0]

        for i in range(1, len(strs)):
            j = 0
            while j < min(len(prefix), len(strs[i])):
                if prefix[j] != strs[i][j]:
                    break
                j += 1
            prefix = prefix[:j]
        return prefix

# Time -> O(n*m) (n is length of shortest string, m is num of strings)
# Space -> O(1)
        
